esmvalcore/config/_config_object.py

```diff
@@ -431,6 +434,39 @@ def start_session(self, name: str) -> Session:
             session = Session(config=self.copy(), name=name)
         return session
 
+    def update_from_dirs(self, dirs: Iterable[str | Path]) -> None:
+        """Update configuration object from directories.
+
+        This will first search for all YAML files within the given directories
+        and merge them together using :func:`dask.config.collect` (if identical
+        values are provided in multiple files, the value from the last file
+        will be used).  Then, the current configuration is merged with these
+        new configuration options using :func:`dask.config.merge` (new values
+        are preferred over old values). Nested objects are properly considered;
+        see :func:`dask.config.update` for details.
+
+        Note
+        ----
+        Just like :func:`dask.config.collect`, this silently ignores
+        non-existing directories.
+
+        Parameters
+        ----------
+        dirs:
+            A list of directories to search for YAML configuration files.
+
+        Raises
+        ------
+        esmvalcore.exceptions.InvalidConfigParameter
+            Invalid configuration option given.
+
+        """
+        new_config_dict = self._get_config_dict_from_dirs(dirs)
+        merged_config_dict = dask.config.merge(self, new_config_dict)
```

Comment:
when this throws a wobbly, do you want to catch and release that or filter it through our own custom exception? Just looked at both `merge` and `update` from Dask, and they're really basic, and not account for any possible exceptions, so if there are exceptions, then those will be cryptic AF

Reply:
When there's an invalid option given a `InvalidConfigParameter` exception is raised in the following line (`self.update()`) via

https://github.com/ESMValGroup/ESMValCore/blob/5b283729c37879c3831ebb1428d056b9d9d7c05f/esmvalcore/config/_validated_config.py#L63-L77

Reply:
ah my apologies Manu, too many GH invocations and tags - what can I say, am popular :grinning: Yes, am always happy with your clarifications :beer: